src/CoreFx.Private.TestUtilities/src/System/PlatformDetection.cs

```diff
@@ -41,6 +41,8 @@ public static partial class PlatformDetection
 
         public static bool IsDomainJoinedMachine => !Environment.MachineName.Equals(Environment.UserDomainName, StringComparison.OrdinalIgnoreCase);
 
+        public static bool IsNotGCForVeryLargeObjects => true;
```

Comment:
What is the definition of the capability that this is controlling?

Reply:
It should be something like https://docs.microsoft.com/en-us/dotnet/framework/configure-apps/file-schema/runtime/gcallowverylargeobjects-element on netcore

Reply:
On .NET Framework, these tests will pass both with `gcAllowVeryLargeObjects=false` and `gcAllowVeryLargeObjects=true`.

Reply:
I wrote like I don't know what blocks coreclr to allocate collections with more than int.MaxValue items, happy to tweak the name to fit that

Reply:
.NET Framework/Core have documented limits on maximum array sizes. From https://docs.microsoft.com/en-us/dotnet/framework/configure-apps/file-schema/runtime/gcallowverylargeobjects-element :

    The maximum index in any single dimension is 2,147,483,591 (0x7FFFFFC7) for byte arrays and arrays of single-byte structures, and 2,146,435,071 (0X7FEFFFFF) for other types.

These limits are less than Int32.MaxValue for defense-in-depth security (to reduce changes of integer overflows causing security bugs), and to allow certain types of bounds-check elimination optimizations and loop unrolling (the JIT can assume that `array.Length + small_constant` won't overflow).

Reply:
Thanks, renamed the property. I didn't know coreclr throws OutOfMemory when indexing into an already allocated array.

Reply:
> I didn't know coreclr throws OutOfMemory when indexing into an already allocated array.

OutOfMemory exception is thrown by `new` when the array is allocated , not when you index into it.

Reply:
@jkotas I have to say this is getting more confusing as the documentation seems to be .NET Framework specific.

```c#
var a = new byte[(uint)int.MaxValue + 1]; // This throws System.OverflowException  Array dimensions exceeded supported range

var b = new byte[int.MaxValue - 100];
Array.Resize (ref b, int.MaxValue);   // This throws OutOfMemoryException 
```

Shouldn't the first one just work, coreclr has LargeObjects enabled by default since https://github.com/dotnet/coreclr/pull/8853/ and you said

>OutOfMemory exception is thrown by new when the array is allocated , not when you index into it.

The second one is why the tests have OOM expectation but should that throw the exception with `Array dimensions exceeded supported range` instead?

Reply:
The behavior of `new int[len]` is:

- `0 <= len && len <= MAX_LENGTH` - Succeeds, assuming there is enough memory
- `MAX_SIZE < len && len <= Int32.MaxInt` - throw OutOfMemoryException.
- `len < 0 || len > Int32.MaxInt` - throw OverflowException 


On .NET Core, and on .NET Framework with gcAllowVeryLargeObjects=true, `MAX_LENGTH` is `sizeof(T) > 1 ? 0X7FEFFFFF : 0x7FFFFFC7` 
On .NET Framework with gcAllowVeryLargeObjects=false, `MAX_LENGTH` is approximately `0x7FFFFFC7 / sizeof(T)` .